src/prefect/cli/cloud.py

```diff
@@ -170,53 +376,48 @@ async def login(
         except httpx.HTTPStatusError as exc:
             exit_with_error(f"Error connecting to Prefect Cloud: {exc!r}")
 
-    profiles = load_profiles()
-
-    for profile_name in profiles:
-        if key == profiles[profile_name].settings.get(PREFECT_API_KEY):
-            profiles.set_active(profile_name)
-            save_profiles(profiles)
-            with prefect.context.use_profile(profile_name):
-                current_workspace = get_current_workspace(workspaces)
-
-                if workspace_handle is not None:
-                    if workspace_handle not in workspace_handle_details:
-                        exit_with_error(f"Workspace {workspace_handle!r} not found.")
-
-                    update_current_profile(
-                        {
-                            PREFECT_API_URL: build_url_from_workspace(
-                                workspace_handle_details[workspace_handle]
-                            )
-                        }
-                    )
-                    current_workspace = workspace_handle
-
-            exit_with_success(
-                f"Logged in to Prefect Cloud using profile {profile_name!r}.\n"
-                f"Workspace is currently set to {current_workspace!r}. "
-                f"The workspace can be changed using `prefect cloud workspace set`."
+    if workspace_handle:
+        # Search for the given workspace
+        for workspace in workspaces:
+            if workspace.handle == workspace_handle:
+                break
+        else:
+            exit_with_error(f"Workspace {workspace_handle!r} not found.")
+    else:
+        # Prompt a switch if the number of workspaces is greater than one
+        prompt_switch_workspace = len(workspaces) > 1
+
+        current_workspace = get_current_workspace(workspaces)
+
+        # Confirm that we want to switch if the current profile is already logged in
+        if (
+            current_profile_is_logged_in or current_workspace is not None
+        ) and prompt_switch_workspace:
+            app.console.print(
+                f"You are currently using workspace {current_workspace.handle!r}."
```

Comment:
```suggestion
                f"You're logged in to workspace {current_workspace.handle!r}."
```

Reply:
I thought we wanted to treat "logged in" and workspace selection separate? You are _logged in_ to your account. You are _using_ a given workspace.